neoemf-core/src/main/java/fr/inria/atlanmod/neoemf/core/internal/LazyStoreList.java

```diff
@@ -0,0 +1,193 @@
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nonnull;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import static java.util.Objects.isNull;
+
+/**
+ * A {@link List} representing a multi-valued feature which behaves as a proxy and that delegates its operations to the
+ * associated {@link Store}.
+ *
+ * @param <E> the type of elements in this list
+ *
+ * @see PersistentEObject#eStore()
+ */
+@ParametersAreNonnullByDefault
+public class LazyStoreList<E> extends EStoreEObjectImpl.BasicEStoreEList<E> {
+
+    @SuppressWarnings("JavaDoc")
+    private static final long serialVersionUID = 2630358403343923944L;
+
+    /**
+     * The object that holds the store.
+     */
+    @Nonnull
+    private final transient Storable storable;
+
+    /**
+     * Constructs a new {@code LazyStoreList}.
+     *
+     * @param owner   the owner the {@code feature}
+     * @param feature the feature associated with this list
+     */
+    public LazyStoreList(PersistentEObject owner, EStructuralFeature feature) {
+        super(owner, feature);
+        this.storable = owner;
+    }
+
+    @Nonnull
+    @Override
+    protected StoreAdapter eStore() {
+        return storable.eStore();
+    }
+
+    // region Delegating methods
+
+    @Override
+    protected void delegateAdd(Object object) {
+        delegateAdd(InternalEObject.EStore.NO_INDEX, object);
+    }
+
+    @Override
+    protected boolean delegateContainsAll(Collection<?> collection) {
+        if (isNull(collection) || collection.isEmpty()) {
+            return false;
+        }
+
+        if (collection.size() <= 1) {
+            return contains(collection.iterator().next());
+        }
+
+        return eStore().getAll(owner, eStructuralFeature).containsAll(collection);
+    }
+
+    @Override
+    // FIXME Does not resolve
+    protected boolean delegateEquals(Object object) {
+        if (object == this) {
+            return true;
+        }
+
+        if (!List.class.isInstance(object)) {
+            return false;
+        }
+
+        List<?> list = List.class.cast(object);
+        return list.size() == delegateSize()
+                && list.equals(eStore().getAll(owner, eStructuralFeature));
+    }
+
+    @Override
+    protected String delegateToString() {
+        return eStore().getAll(owner, eStructuralFeature)
+                .stream()
+                .map(String::valueOf)
+                .collect(Collectors.joining(", ", "[", "]"));
+    }
+
+    // endregion
+
+    @Nonnull
+    @Override
+    public Object[] toArray() {
+        return delegateToArray();
+    }
+
+    @Nonnull
+    @Override
+    public <T> T[] toArray(T[] array) {
+        return delegateToArray(array);
+    }
+
+    @Override
+    // FIXME Does not resolve
+    public boolean contains(Object object) {
+        return delegateContains(object);
+    }
+
+    @Override
+    // FIXME Does not resolve
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1134)